app/controllers/concerns/recovery_code_concern.rb

```diff
@@ -0,0 +1,12 @@
+module RecoveryCodeConcern
+  delegate :active_profile, to: :current_user
+
+  def create_new_code
+    if active_profile.present?
+      Pii::ReEncryptor.new(current_user, user_session).perform
+      active_profile.recovery_code
```

Comment:
I've noticed that we have a few service classes that return something. In general, I think of service classes as places to *do* something. Having a return value that matters gives the object two responsibilities, which violates Single Responsibility Principle. I think that the logic as-is is pretty clear.